go/vt/vtctl/workflow/lookup.go

```diff
@@ -0,0 +1,536 @@
+/*
+Copyright 2024 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package workflow
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"strings"
+
+	"golang.org/x/exp/maps"
+	"google.golang.org/protobuf/proto"
+
+	"vitess.io/vitess/go/sqlescape"
+	"vitess.io/vitess/go/vt/logutil"
+	"vitess.io/vitess/go/vt/schema"
+	"vitess.io/vitess/go/vt/sqlparser"
+	"vitess.io/vitess/go/vt/topo"
+	"vitess.io/vitess/go/vt/vtctl/schematools"
+	"vitess.io/vitess/go/vt/vterrors"
+	"vitess.io/vitess/go/vt/vtgate/vindexes"
+	"vitess.io/vitess/go/vt/vttablet/tmclient"
+
+	tabletmanagerdatapb "vitess.io/vitess/go/vt/proto/tabletmanagerdata"
+	vschemapb "vitess.io/vitess/go/vt/proto/vschema"
+	vtctldatapb "vitess.io/vitess/go/vt/proto/vtctldata"
+	vtrpcpb "vitess.io/vitess/go/vt/proto/vtrpc"
+)
+
+// lookup is responsible for performing actions related to lookup vindexes.
+type lookup struct {
```

Comment:
This is a new struct that has a subset of the workflow server's members, and it's not specific to lookup vindexes at all. Can you help me understand what value this provides? 

Otherwise this LGTM. Thanks!

Reply:
Sure, the idea for defining `lookup` struct was to differentiate the helper functions related to `LookupVindex` from `Server` (as of now `lookup` just differentiates `LookupVindexCreate` helper functions in this PR), this is similar to what was done in #17092 by defining `workflowFetcher`. We can always include more functions (related to lookup vindexes) in follow-ups. Please let me know if you think this can be improved further. Thanks for the review!

Reply:
OK, so I take that to mean this is really for a human reader/developer to try and make the code easier to reason about and manage as lookup vindex related workflow code is in its own specific file and has its own specific method receiver.

In that case, IMO we should do this:
 1. Go from workflow/lookup.go to workflow/lookup_vindex.go
 2. Go from workflow.lookup.prepareCreateLookup to workflow.lookupVindex.prepareCreate (the type becomes lookupVindex and we don't need that lookupvindex context in the function name as its encapsulated in the receiver type

Lookup is a very generic term and it's not otherwise obvious to me how it adds distinction, separation, discoverability, and clarity specifically around the VReplication backfilling work for Lookup Vindexes via the LookupVindex client command and related RPCs. A [lookup vindex](https://vitess.io/docs/reference/features/vindexes/#lookup-vindex-types) is a specific concept in Vitess, and the client command we're processing in the workflow server is [`LookupVindex`](https://vitess.io/docs/21.0/reference/programs/vtctldclient/vtctldclient_lookupvindex/) (also [here](https://vitess.io/docs/21.0/user-guides/vschema-guide/backfill-vindexes/) and [here](https://vitess.io/docs/reference/vreplication/lookupvindex/)). Otherwise I would think that workflow/lookup was about code related to looking up workflows. Does this all make sense?

Thanks!